packages/xrpl/test/createMockRippled.ts

```diff
@@ -47,29 +45,62 @@ export interface PortResponse extends BaseResponse {
  * We mock out WebSocketServer in these tests and add a lot of custom
  * properties not defined on the normal WebSocketServer object.
  */
-// eslint-disable-next-line @typescript-eslint/no-explicit-any -- typing is too complicated otherwise
-type MockedWebSocketServer = any
 
-// eslint-disable-next-line @typescript-eslint/promise-function-async -- Not a promise that's returned
+export type MockedWebSocketServer = WebSocketServer &
+  EventEmitter2 & {
+    responses: Record<string, unknown>
+    suppressOutput: boolean
+    socket: WebSocket
+    addResponse: (
+      command: string,
+      response:
+        | Response
+        | ErrorResponse
+        | ((r: Request) => Response | ErrorResponse | Record<string, unknown>)
+        | Record<string, unknown>,
+    ) => void
+    getResponse: (request: Request) => Record<string, unknown>
+    testCommand: (
+      conn: WebSocket,
+      request: {
+        id: string | number
+        data: {
+          closeServerAndReopen: number
+          disconnectIn: number
+          openOnOtherPort: boolean
+          unrecognizedResponse: boolean
+          closeServer: boolean
+          delayedResponseIn: number
+        }
+      },
+    ) => void
+  }
+
+export function destroyMockRippled(server: MockedWebSocketServer): void {
+  server.removeAllListeners()
+  server.close()
+}
+
 export default function createMockRippled(port: number): MockedWebSocketServer {
   const mock = new WebSocketServer({ port }) as MockedWebSocketServer
   Object.assign(mock, EventEmitter2.prototype)
 
   mock.responses = {}
   mock.suppressOutput = false
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- Typing is too complicated otherwise
-  mock.on('connection', function (this: MockedWebSocketServer, conn: any) {
+  mock.on('connection', function (this: MockedWebSocketServer, conn) {
     this.socket = conn
-    conn.on('message', function (requestJSON: string) {
+    conn.on('message', function (requestJSON) {
```

Comment:
Why are these no longer typed?

Reply:
String isn't the real type, it's `WebSocket.RawData` and is inferred